src/lib/containers.ts

```diff
@@ -0,0 +1,154 @@
+import * as Logging from "../logging"
+const logger = new Logging.Logger("excmd")
+
+/** Creates a container from the specified parameters. Unlike the official Mozilla addon, does not
+ *  allow multiple  containers with the same (name, color, icon) constraint.
+ *  @param containerName  The container name.
+ *  @param containerColor  The container color, limited to browser.contextualIdentities.IdentityColor.
+ *  @param containerIcon  The container icon, limited to browser.contextualIdentities.IdentityIcon.
+ */
+export async function containerCreate(
+    name: string,
+    color: string,
+    icon: string,
+) {
+    let container = containerFromString(name, color, icon)
+    if (await containerExists(name)) {
+        logger.error("[containerCreate] Container already exists, aborting.")
+        logger.debug(container)
+    } else {
+        browser.contextualIdentities
+            .create(container)
```

Comment:
You should make sure that this throws an error if the container can't be created (e.g, if an invalid colour is asked for.)